__tests__/json_decode_test.ml

```diff
@@ -9,121 +45,65 @@ describe "boolean" (fun () ->
-    expectFn bool Encode.null |> toThrow);
-  test "array" (fun () ->
-    expectFn bool (Encode.array [||]) |> toThrow);
-  test "object" (fun () ->
-    expectFn bool (Encode.object_ @@ Js.Dict.empty ()) |> toThrow);
-
-  test "boolean - false" (fun () ->
+  test "bool - false" (fun () ->
     expect @@ bool (Encode.boolean Js.false_) |> toEqual false);
+    
+  Test.throws bool [Float; Int; String; Null; Array; Object];
 );
 
 describe "float" (fun () ->
   let open Json in
   let open! Decode in
 
-  test "boolean" (fun () ->
-    expectFn float (Encode.boolean Js.true_) |> toThrow);
   test "float" (fun () ->
     expect @@ float (Encode.float 1.23) |> toEqual 1.23);
   test "int" (fun () ->
     expect @@ float (Encode.int 23) |> toEqual 23.);
-  test "string" (fun () ->
-    expectFn float (Encode.string "test") |> toThrow);
-  test "null" (fun () ->
-    expectFn float Encode.null |> toThrow);
-  test "array" (fun () ->
-    expectFn float (Encode.array [||]) |> toThrow);
-  test "object" (fun () ->
-    expectFn float (Encode.object_ @@ Js.Dict.empty ()) |> toThrow);
+  
+  Test.throws float [Bool; String; Null; Array; Object;];
 );
 
 describe "int" (fun () ->
   let open Json in
   let open! Decode in
 
-  test "boolean" (fun () ->
-    expectFn int (Encode.boolean Js.true_) |> toThrow);
-  test "float" (fun () ->
-    expectFn int (Encode.float 1.23) |> toThrow);
   test "int" (fun () ->
     expect @@ int (Encode.int 23) |> toEqual 23);
-  test "string" (fun () ->
-    expectFn int (Encode.string "test") |> toThrow);
-  test "null" (fun () ->
-    expectFn int Encode.null |> toThrow);
-  test "array" (fun () ->
-    expectFn int (Encode.array [||]) |> toThrow);
-  test "object" (fun () ->
-    expectFn int (Encode.object_ @@ Js.Dict.empty ()) |> toThrow);
+  
+  Test.throws int [Bool; Float; String; Null; Array; Object];
 );
 
 describe "string" (fun () ->
   let open Json in
   let open! Decode in
 
-  test "boolean" (fun () ->
-    expectFn string (Encode.boolean Js.true_) |> toThrow);
-  test "float" (fun () ->
-    expectFn string (Encode.float 1.23) |> toThrow);
-  test "int" (fun () ->
-    expectFn string (Encode.int 23) |> toThrow);
   test "string" (fun () ->
     expect @@ string (Encode.string "test") |> toEqual "test");
-  test "null" (fun () ->
-    expectFn string Encode.null |> toThrow);
-  test "array" (fun () ->
-    expectFn string (Encode.array [||]) |> toThrow);
-  test "object" (fun () ->
-    expectFn string (Encode.object_ @@ Js.Dict.empty ()) |> toThrow);
+
+  Test.throws string [Bool; Float; Int; Null; Array; Object];
 );
 
 describe "nullable" (fun () ->
   let open Json in
   let open! Decode in
 
-  test "boolean -> int" (fun () ->
-    expectFn (nullable int) (Encode.boolean Js.true_) |> toThrow);
-  test "float -> int" (fun () ->
-    expectFn (nullable int) (Encode.float 1.23) |> toThrow);
+  Test.throws (nullable int) [Bool; Float; String; Array; Object];
+  Test.throws (nullable boolean) [Int];
```

Comment:
The ordering of the tests here is inconsistent with the tests above. `Test.throws` should follow the positive tests.

Also, it would be nice to have an (optional) name prefix and/or suffix argument to `Test.throws`, or some other way to customize the name, to describe and differentiate them better.